### PR TITLE
Fix auto configuration

### PR DESCRIPTION
**Pull request by araithel**

### CODE REVIEW

File: picturesafe-search-spring-boot-demo/src/main/java/de/picturesafe/search/starterdemo/StarterDemoApplication.java [hunk 1]
```diff
@@ -36,9 +40,25 @@ public static void main(String[] args) {
...
+        };
+    }
+
+    private void logBeanDefinitions(ApplicationContext ctx) {
+
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Running Spring Boot app with following bean definitions:");
+            final String[] beanNames = ctx.getBeanDefinitionNames();
+            Arrays.sort(beanNames);
+            for (String beanName : beanNames) {
```
Comment (jfrommann) [inline picturesafe-search-spring-boot-demo/src/main/java/de/picturesafe/search/starterdemo/StarterDemoApplication.java:59]: Checkstyle: missing final

File: picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public QueryConfiguration queryConfiguration() {
     }
 
     @Bean
-    @ConditionalOnMissingBean
+    @ConditionalOnMissingBean(name = "queryFactories")
     public List<QueryFactory> queryFactories(FulltextQueryFactory fulltextQueryFactory,
                                              OperationExpressionQueryFactory operationExpressionQueryFactory,
                                              NestedQueryFactory nestedQueryFactory) {
```
Comment (jfrommann) [inline picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java:64]: Please extend this method to the current implementation in DefaultQueryConfiguration. The order of the query factories is important for the list. We should think about refactoring this.

File: picturesafe-search-spring-boot-autoconfigure/src/test/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoConfigurationTest.java [hunk 3]
```diff
@@ -78,6 +91,16 @@ public void queryConfigurationIsNotNull() {
...
+    @Test
+    public void queryFactoriesIsNotNull() {
+        assertThat(queryFactories).isNotNull();
+    }
+
+    @Test
+    public void filterFactoriesIsNotNull() {
+        assertThat(filterFactories).isNotNull();
+    }
+
```
Comment (jfrommann) [inline picturesafe-search-spring-boot-autoconfigure/src/test/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoConfigurationTest.java:103]: Please check also that the lists are not empty. I had an empty list in the problematic configuration's bean context.

**Review by jfrommann — CHANGES_REQUESTED** [review]

File: picturesafe-search-spring-boot-autoconfigure/src/test/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoConfigurationTest.java (hunk 3 above)
Comment (araithel) [reply picturesafe-search-spring-boot-autoconfigure/src/test/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoConfigurationTest.java:103]: Changed to isNotEmpty

File: picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java (hunk 2 above)
Comment (araithel) [reply picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java:64]: Added RelevanceSortQueryFactory

File: picturesafe-search-spring-boot-demo/src/main/java/de/picturesafe/search/starterdemo/StarterDemoApplication.java (hunk 1 above)
Comment (araithel) [reply picturesafe-search-spring-boot-demo/src/main/java/de/picturesafe/search/starterdemo/StarterDemoApplication.java:59]: I did not see any checkstyle violation...

File: picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java (hunk 2 above)
Comment (jfrommann) [reply picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java:64]: What about FindAllQueryFactory? With "current implementation" i meat the pull request i assigned to you yesterday.

Comment (araithel) [reply picturesafe-search-spring-boot-autoconfigure/src/main/java/de/picturesafe/search/autoconfigure/PicturesafeSearchAutoQueryConfiguration.java]: Merged open pull request from picturesafe-search and added FindAllQueryFactory.

**Review by jfrommann — APPROVED** [review]
Work fine in query generator ;)

**Pull request merged**